src/Validation/gen/Parsers/ValidationsGenerator.TypesParser.cs

```diff
@@ -82,7 +82,7 @@ internal bool TryExtractValidatableType(ITypeSymbol incomingTypeSymbol, WellKnow
 
         visitedTypes.Add(typeSymbol);
 
-        var hasValidationAttributes = HasValidationAttributes(typeSymbol, wellKnownTypes);
+        var hasTypeLevelValidation = HasValidationAttributes(typeSymbol, wellKnownTypes) || HasIValidatableObjectInterface(typeSymbol, wellKnownTypes);
```

Comment:
nit: the name is clearer as hasValidationAttributes. `hasTypeLevelValidation` makes it look like its not considering properties.